cromulent/model.py

```diff
@@ -414,6 +417,22 @@ def __init__(self, ident="", label="", value="", **kw):
 		# Custom post initialization function for autoconstructed classes
 		self._post_init(**kw)
 
+	def __dir__(self):
+		d = dir(self.__class__)
+		d.extend(list(self._list_all_props().keys()))
+		return sorted(d)
+
+	@property
+	def type(self):
+		for c in self._classhier:
+			if c._type:
+				return c.__name__
+
+	@type.setter
+	def type(self, value):
+		raise RequirementError("Must not set 'type' on resources directly")
```

Comment:
Does this match the description of `RequirementError` ("Raised when schema/profile/metadata requirements not met.")?

Reply:
Kinda? It's a requirement of the schema that `type` / `rdf:type` not be randomly reassigned. 

But `AttributeError` is probably closer for this sort of magic.